### PR TITLE
feat(ui): add the machine details numa card

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Add a numa card to the summary tab of the machine details.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- View the machine details for a machine with a numa node.
- You should see a numa card with the details.
- Try with greater than and less than 2 numa nodes and you should see expanded/expandable versions of the card.

## Fixes

Fixes: canonical-web-and-design/MAAS-squad#2110

### CODE REVIEW

File: ui/src/app/machines/views/MachineDetails/MachineSummary/NumaCard/_index.scss [hunk 1]
```diff
@@ -0,0 +1,34 @@
+@mixin NumaCard {
+  .p-numa {
```
Comment (huwshimi) [inline ui/src/app/machines/views/MachineDetails/MachineSummary/NumaCard/_index.scss:2]: Reminder for myself to replace this class with `.numa-card`

File: ui/src/app/machines/views/MachineDetails/MachineSummary/NumaCard/NumaCardDetails/NumaCardDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            },
+            {
+              label: "Storage",
+              value: `${totalStorage.value} ${totalStorage.unit} over ${
+                numaDisks.length
+              } ${pluralize("disk", numaDisks.length)}`,
+            },
+            {
+              label: "Network",
+              value: `${numaInterfaces.length} ${pluralize(
```
Comment (Caleb-Ellis) [inline ui/src/app/machines/views/MachineDetails/MachineSummary/NumaCard/NumaCardDetails/NumaCardDetails.tsx:91]: Same pluralize thing here

File: ui/src/app/machines/views/MachineDetails/MachineSummary/NumaCard/NumaCardDetails/NumaCardDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                "interface",
+                numaInterfaces.length
+              )}`,
+            },
+          ]}
+        />
+      ) : (
+        <span className="p-numa__collapsed-details">
+          {numaNode.cores.length} {pluralize("core", numaNode.cores.length)},
```
Comment (Caleb-Ellis) [inline ui/src/app/machines/views/MachineDetails/MachineSummary/NumaCard/NumaCardDetails/NumaCardDetails.tsx:100]: Just missing a space after this

**Review by Caleb-Ellis — APPROVED** [review]
LGTM! Just some minor suggestions.

The scss is a little bit outdated. Vanilla at least has a standard for icons in buttons now. Maybe we should add a card to revisit the summary tab scss once we're finished?

File: ui/src/app/machines/views/MachineDetails/MachineSummary/NumaCard/NumaCard.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  // TypeScript know we're using the right union type (otherwise it will
+  // complain that numa_nodes doesn't exist on the base machine type).
+  if (!machine || !("numa_nodes" in machine)) {
+    content = <Spinner />;
+  } else {
+    const numaNodes = machine.numa_nodes;
+    content = (
+      <>
+        <strong className="p-muted-heading u-sv1">
+          {numaNodes.length} NUMA {pluralize("node", numaNodes.length)}
```
[the remark this answers is not in the document]
Comment (huwshimi) [reply ui/src/app/machines/views/MachineDetails/MachineSummary/NumaCard/NumaCard.tsx:31]: Nice, thanks.

File: ui/src/app/machines/views/MachineDetails/MachineSummary/NumaCard/NumaCardDetails/NumaCardDetails.tsx (hunk 3 above)
Comment (huwshimi) [reply ui/src/app/machines/views/MachineDetails/MachineSummary/NumaCard/NumaCardDetails/NumaCardDetails.tsx:100]: Done.

File: ui/src/app/machines/views/MachineDetails/MachineSummary/NumaCard/NumaCardDetails/NumaCardDetails.tsx (hunk 2 above)
Comment (huwshimi) [reply ui/src/app/machines/views/MachineDetails/MachineSummary/NumaCard/NumaCardDetails/NumaCardDetails.tsx:91]: Done.

**Comment by huwshimi** [comment]
> LGTM! Just some minor suggestions.
> 
> The scss is a little bit outdated. Vanilla at least has a standard for icons in buttons now. Maybe we should add a card to revisit the summary tab scss once we're finished?

I think the UI is a bit at fault there, it has this kind-of-accordion thing that isn't used elsewhere. I initially tried updating to using the standard button and icon, but it doesn't handle this kind of UI.

Screen Shot 2020-10-28 at 5 25 04 pm

**Pull request merged**